DiscordIan/Module/OpenWeather.cs

```diff
@@ -0,0 +1,317 @@
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+using Discord.Commands;
+using DiscordIan.Model.OpenWeatherMap;
+using DiscordIan.Service;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Options;
+using Microsoft.VisualBasic.CompilerServices;
+
+namespace DiscordIan.Module
+{
+    public class OpenWeather : BaseModule
+    {
+        private readonly IDistributedCache _cache;
+        private readonly FetchJsonService _fetchJsonService;
+        private readonly FetchXMLService _fetchXMLService;
+        private readonly Model.Options _options;
+        private const string Current = "weather";
+        private const string Forecast = "forecast";
+
+        public OpenWeather(IDistributedCache cache,
+            FetchJsonService fetchJsonService,
+            FetchXMLService fetchXMLService,
+            IOptionsMonitor<Model.Options> optionsAccessor)
+        {
+            _cache = cache ?? throw new ArgumentNullException(nameof(cache));
+            _fetchJsonService = fetchJsonService
+                ?? throw new ArgumentNullException(nameof(fetchJsonService));
+            _fetchXMLService = fetchXMLService
+                ?? throw new ArgumentNullException(nameof(fetchXMLService));
+            _options = optionsAccessor.CurrentValue
+                ?? throw new ArgumentNullException(nameof(optionsAccessor));
+        }
+
+        [Command("wz", RunMode = RunMode.Async)]
+        [Summary("Look up current weather for a provided address.")]
+        [Alias("weather", "w")]
+        public async Task CurrentAsync([Remainder]
+            [Summary("The address or location for current weather conditions")] string location)
+        {
+            string coords = await _cache.GetStringAsync(location);
+            string locale = coords == null ? null : await _cache.GetStringAsync(coords);
+
+            if (string.IsNullOrEmpty(coords) && string.IsNullOrEmpty(location))
+            {
+                await ReplyAsync("Please provide a location.");
+                return;
+            }
+            Model.Geocodio.Location coordinates = null;
+            Model.Geocodio.AddressComponents address = null;
+            if (coords == null || locale == null)
+            {
+                try
+                {
+                    (coordinates, address) = await GeocodeAddressAsync(location);
+
+                    if (coordinates != null && address != null)
+                    {
+                        locale = address.City + ", " + address.State;
+                        string latlong = String.Format("{0},{1}", coordinates.Lat.ToString(), coordinates.Lng.ToString());
+
+                        await _cache.SetStringAsync(location, latlong,
+                            new DistributedCacheEntryOptions { AbsoluteExpirationRelativeToNow = TimeSpan.FromHours(4) });
```

Comment:
Prefer to wrap at 100 characters for readability.

Reply:
DISAGREEEEE